src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -311,77 +309,63 @@ protected bool IsFullyPreserved (TypeDefinition type)
 			return false;
 		}
 
-		internal void MarkEntireType (TypeDefinition type, bool includeBaseAndInterfaceTypes, in DependencyInfo reason)
+		internal void MarkEntireType (TypeDefinition type, in DependencyInfo reason)
 		{
 			// Prevent cases where there's nothing on the stack (can happen when marking entire assemblies)
 			// In which case we would generate warnings with no source (hard to debug)
 			using var _ = _scopeStack.CurrentScope.Origin.MemberDefinition == null ? _scopeStack.PushScope (new MessageOrigin (type)) : null;
 
-			MarkEntireTypeInternal (type, includeBaseAndInterfaceTypes, reason);
+			MarkEntireTypeInternal (type, reason);
 		}
 
-		void MarkEntireTypeInternal (TypeDefinition type, bool includeBaseAndInterfaceTypes, in DependencyInfo reason)
+		void MarkEntireTypeInternal (TypeDefinition type, in DependencyInfo reason)
```

Comment:
Since you are modifying these methods, is there a reason for keeping both `MarkEntireType` and `MarkEntireTypeInternal`?

Reply:
I think this comment from `MarkEntireType` is still relevant and unnecessary for the recursive call: https://github.com/mono/linker/blob/ffe71535ba927280e6e93db0ca04dd99bde07637/src/linker/Linker.Steps/MarkStep.cs#L316-L318

Reply:
I guess it doesn't matter much. But if we had it on `MarkEntireTypeInternal` it'd work the same way, since on a second we are guaranteed to have something on the stack (and so the `IDisposable` will be null).

Reply:
Good point, I took the suggestion. Thanks!